meerk40t/core/node/op_cut.py

```diff
@@ -274,9 +277,15 @@ def preprocess(self, context, matrix, plan):
             except ValueError:
                 pass
         native_mm = abs(complex(*matrix.transform_vector([0, UNITS_PER_MM])))
+        if self.kerf is None:
+            self.kerf = 0
         self.settings["native_mm"] = native_mm
         self.settings["native_speed"] = self.speed * native_mm
         self.settings["native_rapid_speed"] = self.rapid_speed * native_mm
+        # We need to establish the native device resolution,
+        # as kerf is given in scene space but needs to be passed on in device space
+        device = context.device
+        self._device_factor = 1 / (device.native_scale_x**2 + device.native_scale_y**2)**0.5
```

Comment:
This is always easier to do in complex numbers. `1/abs(complex(device.native_scale_x, device.native_scale_y))`